dotCMS/src/main/java/com/dotmarketing/db/DotRunnableThread.java

```diff
@@ -0,0 +1,66 @@
+package com.dotmarketing.db;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import com.dotmarketing.business.DotStateException;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.util.Config;
+
+public class DotRunnableThread extends Thread {
+
+    final List<DotRunnable> listeners;
+
+    public DotRunnableThread(final List<DotRunnable> listeners) {
+
+        this.listeners = listeners;
+    }
+
+    @Override
+    public void run() {
+        try {
+            LocalTransaction.wrap(this::internalRunner);
+        } catch (DotDataException dde) {
+            throw new DotStateException(dde);
+        }
+    }
+
+    private void internalRunner() {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 20 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)